internal/cli/tiltfile_result.go

```diff
@@ -0,0 +1,88 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	tiltanalytics "github.com/windmilleng/tilt/internal/analytics"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+	"github.com/windmilleng/tilt/pkg/logger"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+const TiltfileErrExitCode = 5
+
+type tiltfileResultDeps struct {
+	tfl tiltfile.TiltfileLoader
+}
+
+func newTiltfileResultDeps(tfl tiltfile.TiltfileLoader) tiltfileResultDeps {
+	return tiltfileResultDeps{
+		tfl: tfl,
+	}
+}
+
+func newTiltfileResultCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tiltfile-result",
+		Short: "exec the tiltfile and print results as json",
+
+		Run: tiltfileResultPrintJSON,
+	}
+
+	cmd.Flags().StringVar(&fileName, "file", tiltfile.FileName, "Path to Tiltfile")
+
+	return cmd
+}
+
+func tiltfileResultPrintJSON(cmd *cobra.Command, args []string) {
+	l := logger.NewLogger(logger.DebugLvl, os.Stdout)
```

Comment:
can you elaborate a bit on why you didn't make this a normal tilt command that uses preCommand, like tilt up / tilt down? the way all these things interact is pretty subtle

Reply:
Oh I assumed `preCommand` has a bunch of `up`/`down`-specific stuff. Can do!